src/tasks/service.rs

```diff
@@ -826,12 +750,8 @@ async fn load_wasm_modules(wasm_dir: &Path, provider_types: Vec<String>) -> Wasm
 
     provider_types.into_iter().zip(runtimes).collect()
 }
-fn get_protocol_version(provider_type: &str) -> u8 {
-    if V1_PROVIDERS.contains(&provider_type) {
-        1
-    } else {
-        2
-    }
+fn get_protocol_version(_: &str) -> u8 {
+    2
 }
```

Comment:
Is it possible to inline all the uses of that function? Now that we might have multiple versions of providers in the wild, the "protocol detection" logic should happen entirely from the communication between the provider and FPD anyways, so we should never need that `get_protocol_version(hardcoded_name: &str)` function again

Reply:
Implemented in ce7927dc145b8a4ef07cdabc47f8fc160dfe0d6b